wdl-lint/src/rules/misplaced_lint_directive.rs

```diff
@@ -0,0 +1,133 @@
+//! A lint rule for flagging misplaced lint directives.
+
+use wdl_ast::AstToken;
+use wdl_ast::Comment;
+use wdl_ast::Diagnostic;
+use wdl_ast::Diagnostics;
+use wdl_ast::Document;
+use wdl_ast::Span;
+use wdl_ast::SupportedVersion;
+use wdl_ast::SyntaxElement;
+use wdl_ast::SyntaxKind;
+use wdl_ast::ToSpan;
+use wdl_ast::VisitReason;
+use wdl_ast::Visitor;
+use wdl_ast::EXCEPT_COMMENT_PREFIX;
+
+use crate::rules;
+use crate::Rule;
+use crate::Tag;
+use crate::TagSet;
+
+/// The identifier for the unknown rule rule.
+const ID: &str = "MisplacedLintDirective";
+
+/// Creates an "unknown rule" diagnostic.
+fn misplaced_lint_directive(
+    id: &str,
+    span: Span,
+    wrong_element: &SyntaxElement,
+    exceptable_nodes: &[SyntaxKind],
+) -> Diagnostic {
+    let locations = exceptable_nodes
+        .iter()
+        .map(|node| node.describe())
+        .collect::<Vec<_>>()
+        .join(", ");
+
+    Diagnostic::note(format!(
+        "lint directive `{id}` has no effect above: {elem}",
```

Comment:
Since we're not printing a list here:

```suggestion
        "lint directive `{id}` has no effect above {elem}",
```